File: Examples/UI/WinForm/VB.NET/ProgressBars/NWaitingBarUC.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Windows.Forms

Imports Nevron.UI.WinForm.Controls
Imports Nevron.Examples.Framework.WinForm

Namespace Nevron.Examples.UI.WinForm
	''' <summary>
	''' Summary description for NWaitingBarUC.
	''' </summary>
	Public Class NWaitingBarUC
		Inherits NExampleUserControl
		#Region "Constructor"

		Public Sub New(ByVal f As MainForm)
			MyBase.New(f)
			InitializeComponent()
			m_iSuspendUpdate = 0
		End Sub


		#End Region

		#Region "Implementation"

		Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If Not components Is Nothing Then
					components.Dispose()
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub
		Public Overrides Sub Initialize()
			MyBase.Initialize ()

			m_iSuspendUpdate += 1

			m_StyleCombo.FillFromEnum(GetType(Nevron.UI.WinForm.Controls.ProgressBarStyle), False)
			m_StyleCombo.SelectedItem = Nevron.UI.WinForm.Controls.ProgressBarStyle.Solid

			m_OrientationCombo.FillFromEnum(GetType(Orientation), False)
			m_OrientationCombo.SelectedItem = Orientation.Horizontal
			nhScrollBar1.Value = 100 - nWaitingBar1.Properties.Interval

			m_WaitSizeNumeric.Value = nWaitingBar1.Properties.WaitSize
			m_StepNumeric.Value = nWaitingBar1.Properties.Step

			textEdit.Text = "Installing... Please, wait!"

			m_iSuspendUpdate -= 1

			Me.nWaitingBar1.BeginWait()
		End Sub


		#End Region

		#Region "Event Handlers"

		Private Sub m_StyleCombo_SelectedIndexChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles m_StyleCombo.SelectedIndexChanged
			If m_iSuspendUpdate <> 0 Then
				Return
			End If
			Dim style As Nevron.UI.WinForm.Controls.ProgressBarStyle = CType(m_StyleCombo.SelectedItem, Nevron.UI.WinForm.Controls.ProgressBarStyle)
			nWaitingBar1.Properties.Style = style
		End Sub
		Private Sub m_OrientationCombo_SelectedIndexChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles m_OrientationCombo.SelectedIndexChanged
			If m_iSuspendUpdate <> 0 Then
				Return
			End If
			Dim orientation As Orientation = CType(m_OrientationCombo.SelectedItem, Orientation)
			nWaitingBar1.Properties.Orientation = orientation
		End Sub
		Private Sub m_BorderButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles m_BorderButton.Click
			nWaitingBar1.Border.ShowEditor()
		End Sub
		Private Sub m_PaletteButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles m_PaletteButton.Click
			nWaitingBar1.Palette.ShowEditor()
		End Sub
		Private Sub m_BeginWaitButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles m_BeginWaitButton.Click
			nWaitingBar1.BeginWait()
		End Sub
		Private Sub m_EndWaitButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles m_EndWaitButton.Click
			nWaitingBar1.EndWait()
		End Sub

		Private Sub nhScrollBar1_ValueChanged(ByVal sender As Object, ByVal e As Nevron.UI.WinForm.Controls.ScrollBarEventArgs) Handles nhScrollBar1.ValueChanged
			If m_iSuspendUpdate <> 0 Then
				Return
			End If
			nWaitingBar1.Properties.Interval = 100 - nhScrollBar1.Value
		End Sub
		Private Sub m_WaitSizeNumeric_ValueChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles m_WaitSizeNumeric.ValueChanged
			If m_iSuspendUpdate <> 0 Then
				Return
			End If
			nWaitingBar1.Properties.WaitSize = CInt(Fix(m_WaitSizeNumeric.Value))
		End Sub
		Private Sub m_IncrementButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles m_IncrementButton.Click
			nWaitingBar1.Properties.Position += 1
		End Sub
		Private Sub m_StepNumeric_ValueChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles m_StepNumeric.ValueChanged
			If m_iSuspendUpdate <> 0 Then
				Return
			End If
			nWaitingBar1.Properties.Step = CInt(Fix(m_StepNumeric.Value))
		End Sub

		Private Sub textEdit_TextChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles textEdit.TextChanged
			nWaitingBar1.Properties.Text = textEdit.Text
		End Sub


		#End Region

		#Region "Component Designer generated code"
		''' <summary> 
		''' Required method for Designer support - do not modify 
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
			Me.nWaitingBar1 = New Nevron.UI.WinForm.Controls.NWaitingBar()
			Me.label4 = New System.Windows.Forms.Label()
			Me.m_OrientationCombo = New Nevron.UI.WinForm.Controls.NComboBox()
			Me.m_PaletteButton = New Nevron.UI.WinForm.Controls.NButton()
			Me.label2 = New System.Windows.Forms.Label()
			Me.m_StyleCombo = New Nevron.UI.WinForm.Controls.NComboBox()
			Me.m_BorderButton = New Nevron.UI.WinForm.Controls.NButton()
			Me.m_BeginWaitButton = New Nevron.UI.WinForm.Controls.NButton()
			Me.m_EndWaitButton = New Nevron.UI.WinForm.Controls.NButton()
			Me.nhScrollBar1 = New Nevron.UI.WinForm.Controls.NHScrollBar()
			Me.label1 = New System.Windows.Forms.Label()
			Me.label3 = New System.Windows.Forms.Label()
			Me.m_WaitSizeNumeric = New Nevron.UI.WinForm.Controls.NNumericUpDown()
			Me.label5 = New System.Windows.Forms.Label()
			Me.m_StepNumeric = New Nevron.UI.WinForm.Controls.NNumericUpDown()
			Me.m_IncrementButton = New Nevron.UI.WinForm.Controls.NButton()
			Me.textEdit = New Nevron.UI.WinForm.Controls.NTextBox()
			Me.label6 = New System.Windows.Forms.Label()
			CType(Me.m_WaitSizeNumeric, System.ComponentModel.ISupportInitialize).BeginInit()
			CType(Me.m_StepNumeric, System.ComponentModel.ISupportInitialize).BeginInit()
			Me.SuspendLayout()
			' 
			' nWaitingBar1
			' 
			Me.nWaitingBar1.Location = New System.Drawing.Point(16, 16)
			Me.nWaitingBar1.Name = "nWaitingBar1"
			Me.nWaitingBar1.Size = New System.Drawing.Size(312, 24)
			Me.nWaitingBar1.TabIndex = 0
			Me.nWaitingBar1.Text = "nWaitingBar1"
			' 
			' label4
			' 
			Me.label4.Location = New System.Drawing.Point(48, 144)
			Me.label4.Name = "label4"
			Me.label4.Size = New System.Drawing.Size(88, 23)
			Me.label4.TabIndex = 35
			Me.label4.Text = "Orientation:"
			Me.label4.TextAlign = System.Drawing.ContentAlignment.MiddleRight
			' 
			' m_OrientationCombo
			' 
			Me.m_OrientationCombo.Location = New System.Drawing.Point(144, 144)
			Me.m_OrientationCombo.Name = "m_OrientationCombo"
			Me.m_OrientationCombo.Size = New System.Drawing.Size(176, 22)
			Me.m_OrientationCombo.TabIndex = 34
			Me.m_OrientationCombo.Text = "nComboBox1"
'			Me.m_OrientationCombo.SelectedIndexChanged += New System.EventHandler(Me.m_OrientationCombo_SelectedIndexChanged);
			' 
			' m_PaletteButton
			' 
			Me.m_PaletteButton.Location = New System.Drawing.Point(232, 304)
			Me.m_PaletteButton.Name = "m_PaletteButton"
			Me.m_PaletteButton.TabIndex = 33
			Me.m_PaletteButton.Text = "Palette..."
'			Me.m_PaletteButton.Click += New System.EventHandler(Me.m_PaletteButton_Click);
			' 
			' label2
			' 
			Me.label2.Location = New System.Drawing.Point(48, 112)
			Me.label2.Name = "label2"
			Me.label2.Size = New System.Drawing.Size(88, 23)
			Me.label2.TabIndex = 32
			Me.label2.Text = "Style:"
			Me.label2.TextAlign = System.Drawing.ContentAlignment.MiddleRight
			' 
			' m_StyleCombo
			' 
			Me.m_StyleCombo.Location = New System.Drawing.Point(144, 112)
			Me.m_StyleCombo.Name = "m_StyleCombo"
			Me.m_StyleCombo.Size = New System.Drawing.Size(176, 22)
			Me.m_StyleCombo.TabIndex = 31
			Me.m_StyleCombo.Text = "nComboBox1"
'			Me.m_StyleCombo.SelectedIndexChanged += New System.EventHandler(Me.m_StyleCombo_SelectedIndexChanged);
			' 
			' m_BorderButton
			' 
			Me.m_BorderButton.Location = New System.Drawing.Point(144, 304)
			Me.m_BorderButton.Name = "m_BorderButton"
			Me.m_BorderButton.TabIndex = 30
			Me.m_BorderButton.Text = "Border..."
'			Me.m_BorderButton.Click += New System.EventHandler(Me.m_BorderButton_Click);
			' 
			' m_BeginWaitButton
			' 
			Me.m_BeginWaitButton.Location = New System.Drawing.Point(336, 16)
			Me.m_BeginWaitButton.Name = "m_BeginWaitButton"
			Me.m_BeginWaitButton.TabIndex = 36
			Me.m_BeginWaitButton.Text = "Begin Wait"
'			Me.m_BeginWaitButton.Click += New System.EventHandler(Me.m_BeginWaitButton_Click);
			' 
			' m_EndWaitButton
			' 
			Me.m_EndWaitButton.Location = New System.Drawing.Point(336, 48)
			Me.m_EndWaitButton.Name = "m_EndWaitButton"
			Me.m_EndWaitButton.TabIndex = 37
			Me.m_EndWaitButton.Text = "End Wait"
'			Me.m_EndWaitButton.Click += New System.EventHandler(Me.m_EndWaitButton_Click);
			' 
			' nhScrollBar1
			' 
			Me.nhScrollBar1.Location = New System.Drawing.Point(144, 176)
			Me.nhScrollBar1.Name = "nhScrollBar1"
			Me.nhScrollBar1.Size = New System.Drawing.Size(176, 17)
			Me.nhScrollBar1.TabIndex = 38
			Me.nhScrollBar1.Text = "nhScrollBar1"
'			Me.nhScrollBar1.ValueChanged += New Nevron.UI.WinForm.Controls.ScrollBarEventHandler(Me.nhScrollBar1_ValueChanged);
			' 
			' label1
			' 
			Me.label1.Location = New System.Drawing.Point(72, 176)
			Me.label1.Name = "label1"
			Me.label1.Size = New System.Drawing.Size(64, 23)
			Me.label1.TabIndex = 39
			Me.label1.Text = "Speed:"
			Me.label1.TextAlign = System.Drawing.ContentAlignment.MiddleRight
			' 
			' label3
			' 
			Me.label3.Location = New System.Drawing.Point(72, 240)
			Me.label3.Name = "label3"
			Me.label3.Size = New System.Drawing.Size(64, 23)
			Me.label3.TabIndex = 41
			Me.label3.Text = "Wait Size:"
			Me.label3.TextAlign = System.Drawing.ContentAlignment.MiddleRight
			' 
			' m_WaitSizeNumeric
			' 
			Me.m_WaitSizeNumeric.Location = New System.Drawing.Point(144, 240)
			Me.m_WaitSizeNumeric.Name = "m_WaitSizeNumeric"
			Me.m_WaitSizeNumeric.Size = New System.Drawing.Size(72, 20)
			Me.m_WaitSizeNumeric.TabIndex = 40
'			Me.m_WaitSizeNumeric.ValueChanged += New System.EventHandler(Me.m_WaitSizeNumeric_ValueChanged);
			' 
			' label5
			' 
			Me.label5.Location = New System.Drawing.Point(72, 272)
			Me.label5.Name = "label5"
			Me.label5.Size = New System.Drawing.Size(64, 23)
			Me.label5.TabIndex = 43
			Me.label5.Text = "Step:"
			Me.label5.TextAlign = System.Drawing.ContentAlignment.MiddleRight
			' 
			' m_StepNumeric
			' 
			Me.m_StepNumeric.Location = New System.Drawing.Point(144, 272)
			Me.m_StepNumeric.Name = "m_StepNumeric"
			Me.m_StepNumeric.Size = New System.Drawing.Size(72, 20)
			Me.m_StepNumeric.TabIndex = 42
'			Me.m_StepNumeric.ValueChanged += New System.EventHandler(Me.m_StepNumeric_ValueChanged);
			' 
			' m_IncrementButton
			' 
			Me.m_IncrementButton.Location = New System.Drawing.Point(336, 80)
			Me.m_IncrementButton.Name = "m_IncrementButton"
			Me.m_IncrementButton.TabIndex = 44
			Me.m_IncrementButton.Text = "Increment"
'			Me.m_IncrementButton.Click += New System.EventHandler(Me.m_IncrementButton_Click);
			' 
			' textEdit
			' 
			Me.textEdit.Location = New System.Drawing.Point(144, 208)
			Me.textEdit.Name = "textEdit"
			Me.textEdit.Size = New System.Drawing.Size(176, 18)
			Me.textEdit.TabIndex = 45
'			Me.textEdit.TextChanged += New System.EventHandler(Me.textEdit_TextChanged);
			' 
			' label6
			' 
			Me.label6.Location = New System.Drawing.Point(72, 208)
			Me.label6.Name = "label6"
			Me.label6.Size = New System.Drawing.Size(64, 23)
			Me.label6.TabIndex = 46
			Me.label6.Text = "Text:"
			Me.label6.TextAlign = System.Drawing.ContentAlignment.MiddleRight
			' 
			' NWaitingBarUC
			' 
			Me.Controls.Add(Me.label6)
			Me.Controls.Add(Me.textEdit)
			Me.Controls.Add(Me.m_IncrementButton)
			Me.Controls.Add(Me.label5)
			Me.Controls.Add(Me.m_StepNumeric)
			Me.Controls.Add(Me.label3)
			Me.Controls.Add(Me.m_WaitSizeNumeric)
			Me.Controls.Add(Me.label1)
			Me.Controls.Add(Me.nhScrollBar1)
			Me.Controls.Add(Me.m_EndWaitButton)
			Me.Controls.Add(Me.m_BeginWaitButton)
			Me.Controls.Add(Me.label4)
			Me.Controls.Add(Me.m_OrientationCombo)
			Me.Controls.Add(Me.m_PaletteButton)
			Me.Controls.Add(Me.label2)
			Me.Controls.Add(Me.m_StyleCombo)
			Me.Controls.Add(Me.m_BorderButton)
			Me.Controls.Add(Me.nWaitingBar1)
			Me.Name = "NWaitingBarUC"
			Me.Size = New System.Drawing.Size(424, 336)
			CType(Me.m_WaitSizeNumeric, System.ComponentModel.ISupportInitialize).EndInit()
			CType(Me.m_StepNumeric, System.ComponentModel.ISupportInitialize).EndInit()
			Me.ResumeLayout(False)

		End Sub
		#End Region

		#Region "Fields"

		Private components As System.ComponentModel.Container = Nothing

		Private nWaitingBar1 As Nevron.UI.WinForm.Controls.NWaitingBar
		Private label4 As System.Windows.Forms.Label
		Private WithEvents m_OrientationCombo As Nevron.UI.WinForm.Controls.NComboBox
		Private WithEvents m_PaletteButton As Nevron.UI.WinForm.Controls.NButton
		Private label2 As System.Windows.Forms.Label
		Private WithEvents m_StyleCombo As Nevron.UI.WinForm.Controls.NComboBox
		Private WithEvents m_BorderButton As Nevron.UI.WinForm.Controls.NButton
		Private WithEvents m_BeginWaitButton As Nevron.UI.WinForm.Controls.NButton
		Private WithEvents m_EndWaitButton As Nevron.UI.WinForm.Controls.NButton
		Private WithEvents nhScrollBar1 As Nevron.UI.WinForm.Controls.NHScrollBar
		Private label1 As System.Windows.Forms.Label
		Private label3 As System.Windows.Forms.Label
		Private WithEvents m_WaitSizeNumeric As Nevron.UI.WinForm.Controls.NNumericUpDown
		Private label5 As System.Windows.Forms.Label
		Private WithEvents m_StepNumeric As Nevron.UI.WinForm.Controls.NNumericUpDown
		Private WithEvents textEdit As Nevron.UI.WinForm.Controls.NTextBox
		Private label6 As System.Windows.Forms.Label
		Private WithEvents m_IncrementButton As Nevron.UI.WinForm.Controls.NButton

		#End Region
	End Class
End Namespace

File: Examples/UI/WinForm/CSharp/ProgressBars/NWaitingBarUC.cs
using System;
using System.Collections;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Windows.Forms;

using Nevron.UI.WinForm.Controls;
using Nevron.Examples.Framework.WinForm;

namespace Nevron.Examples.UI.WinForm
{
	/// <summary>
	/// Summary description for NWaitingBarUC.
	/// </summary>
	public class NWaitingBarUC : NExampleUserControl
	{
		#region Constructor

		public NWaitingBarUC(MainForm f) : base(f)
		{
			InitializeComponent();
			m_iSuspendUpdate = 0;
		}


		#endregion

		#region Implementation

		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}
		public override void Initialize()
		{
			base.Initialize ();

			++m_iSuspendUpdate;

			m_StyleCombo.FillFromEnum(typeof(Nevron.UI.WinForm.Controls.ProgressBarStyle), false);
			m_StyleCombo.SelectedItem = Nevron.UI.WinForm.Controls.ProgressBarStyle.Solid;

			m_OrientationCombo.FillFromEnum(typeof(Orientation), false);
			m_OrientationCombo.SelectedItem = Orientation.Horizontal;
			nhScrollBar1.Value = 100 - nWaitingBar1.Properties.Interval;

			m_WaitSizeNumeric.Value = nWaitingBar1.Properties.WaitSize;
			m_StepNumeric.Value = nWaitingBar1.Properties.Step;

			textEdit.Text = "Installing... Please, wait!";

			--m_iSuspendUpdate;

			this.nWaitingBar1.BeginWait();
		}


		#endregion

		#region Event Handlers

		private void m_StyleCombo_SelectedIndexChanged(object sender, System.EventArgs e)
		{
			if(m_iSuspendUpdate != 0)
				return;
			Nevron.UI.WinForm.Controls.ProgressBarStyle style = (Nevron.UI.WinForm.Controls.ProgressBarStyle)m_StyleCombo.SelectedItem;
			nWaitingBar1.Properties.Style = style;
		}
		private void m_OrientationCombo_SelectedIndexChanged(object sender, System.EventArgs e)
		{
			if(m_iSuspendUpdate != 0)
				return;
			Orientation orientation = (Orientation)m_OrientationCombo.SelectedItem;
			nWaitingBar1.Properties.Orientation = orientation;
		}
		private void m_BorderButton_Click(object sender, System.EventArgs e)
		{
			nWaitingBar1.Border.ShowEditor();
		}
		private void m_PaletteButton_Click(object sender, System.EventArgs e)
		{
			nWaitingBar1.Palette.ShowEditor();
		}
		private void m_BeginWaitButton_Click(object sender, System.EventArgs e)
		{
			nWaitingBar1.BeginWait();
		}
		private void m_EndWaitButton_Click(object sender, System.EventArgs e)
		{
			nWaitingBar1.EndWait();
		}

		private void nhScrollBar1_ValueChanged(object sender, Nevron.UI.WinForm.Controls.ScrollBarEventArgs e)
		{
			if(m_iSuspendUpdate != 0)
				return;
			nWaitingBar1.Properties.Interval = 100 - nhScrollBar1.Value;
		}
		private void m_WaitSizeNumeric_ValueChanged(object sender, System.EventArgs e)
		{
			if(m_iSuspendUpdate != 0)
				return;
			nWaitingBar1.Properties.WaitSize = (int)m_WaitSizeNumeric.Value;
		}
		private void m_IncrementButton_Click(object sender, System.EventArgs e)
		{
			nWaitingBar1.Properties.Position += 1;
		}
		private void m_StepNumeric_ValueChanged(object sender, System.EventArgs e)
		{
			if(m_iSuspendUpdate != 0)
				return;
			nWaitingBar1.Properties.Step = (int)m_StepNumeric.Value;
		}

		private void textEdit_TextChanged(object sender, System.EventArgs e)
		{
			nWaitingBar1.Properties.Text = textEdit.Text;
		}


		#endregion

		#region Component Designer generated code
		/// <summary> 
		/// Required method for Designer support - do not modify 
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.nWaitingBar1 = new Nevron.UI.WinForm.Controls.NWaitingBar();
			this.label4 = new System.Windows.Forms.Label();
			this.m_OrientationCombo = new Nevron.UI.WinForm.Controls.NComboBox();
			this.m_PaletteButton = new Nevron.UI.WinForm.Controls.NButton();
			this.label2 = new System.Windows.Forms.Label();
			this.m_StyleCombo = new Nevron.UI.WinForm.Controls.NComboBox();
			this.m_BorderButton = new Nevron.UI.WinForm.Controls.NButton();
			this.m_BeginWaitButton = new Nevron.UI.WinForm.Controls.NButton();
			this.m_EndWaitButton = new Nevron.UI.WinForm.Controls.NButton();
			this.nhScrollBar1 = new Nevron.UI.WinForm.Controls.NHScrollBar();
			this.label1 = new System.Windows.Forms.Label();
			this.label3 = new System.Windows.Forms.Label();
			this.m_WaitSizeNumeric = new Nevron.UI.WinForm.Controls.NNumericUpDown();
			this.label5 = new System.Windows.Forms.Label();
			this.m_StepNumeric = new Nevron.UI.WinForm.Controls.NNumericUpDown();
			this.m_IncrementButton = new Nevron.UI.WinForm.Controls.NButton();
			this.textEdit = new Nevron.UI.WinForm.Controls.NTextBox();
			this.label6 = new System.Windows.Forms.Label();
			((System.ComponentModel.ISupportInitialize)(this.m_WaitSizeNumeric)).BeginInit();
			((System.ComponentModel.ISupportInitialize)(this.m_StepNumeric)).BeginInit();
			this.SuspendLayout();
			// 
			// nWaitingBar1
			// 
			this.nWaitingBar1.Location = new System.Drawing.Point(16, 16);
			this.nWaitingBar1.Name = "nWaitingBar1";
			this.nWaitingBar1.Size = new System.Drawing.Size(312, 24);
			this.nWaitingBar1.TabIndex = 0;
			this.nWaitingBar1.Text = "nWaitingBar1";
			// 
			// label4
			// 
			this.label4.Location = new System.Drawing.Point(48, 144);
			this.label4.Name = "label4";
			this.label4.Size = new System.Drawing.Size(88, 23);
			this.label4.TabIndex = 35;
			this.label4.Text = "Orientation:";
			this.label4.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
			// 
			// m_OrientationCombo
			// 
			this.m_OrientationCombo.Location = new System.Drawing.Point(144, 144);
			this.m_OrientationCombo.Name = "m_OrientationCombo";
			this.m_OrientationCombo.Size = new System.Drawing.Size(176, 22);
			this.m_OrientationCombo.TabIndex = 34;
			this.m_OrientationCombo.Text = "nComboBox1";
			this.m_OrientationCombo.SelectedIndexChanged += new System.EventHandler(this.m_OrientationCombo_SelectedIndexChanged);
			// 
			// m_PaletteButton
			// 
			this.m_PaletteButton.Location = new System.Drawing.Point(232, 304);
			this.m_PaletteButton.Name = "m_PaletteButton";
			this.m_PaletteButton.TabIndex = 33;
			this.m_PaletteButton.Text = "Palette...";
			this.m_PaletteButton.Click += new System.EventHandler(this.m_PaletteButton_Click);
			// 
			// label2
			// 
			this.label2.Location = new System.Drawing.Point(48, 112);
			this.label2.Name = "label2";
			this.label2.Size = new System.Drawing.Size(88, 23);
			this.label2.TabIndex = 32;
			this.label2.Text = "Style:";
			this.label2.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
			// 
			// m_StyleCombo
			// 
			this.m_StyleCombo.Location = new System.Drawing.Point(144, 112);
			this.m_StyleCombo.Name = "m_StyleCombo";
			this.m_StyleCombo.Size = new System.Drawing.Size(176, 22);
			this.m_StyleCombo.TabIndex = 31;
			this.m_StyleCombo.Text = "nComboBox1";
			this.m_StyleCombo.SelectedIndexChanged += new System.EventHandler(this.m_StyleCombo_SelectedIndexChanged);
			// 
			// m_BorderButton
			// 
			this.m_BorderButton.Location = new System.Drawing.Point(144, 304);
			this.m_BorderButton.Name = "m_BorderButton";
			this.m_BorderButton.TabIndex = 30;
			this.m_BorderButton.Text = "Border...";
			this.m_BorderButton.Click += new System.EventHandler(this.m_BorderButton_Click);
			// 
			// m_BeginWaitButton
			// 
			this.m_BeginWaitButton.Location = new System.Drawing.Point(336, 16);
			this.m_BeginWaitButton.Name = "m_BeginWaitButton";
			this.m_BeginWaitButton.TabIndex = 36;
			this.m_BeginWaitButton.Text = "Begin Wait";
			this.m_BeginWaitButton.Click += new System.EventHandler(this.m_BeginWaitButton_Click);
			// 
			// m_EndWaitButton
			// 
			this.m_EndWaitButton.Location = new System.Drawing.Point(336, 48);
			this.m_EndWaitButton.Name = "m_EndWaitButton";
			this.m_EndWaitButton.TabIndex = 37;
			this.m_EndWaitButton.Text = "End Wait";
			this.m_EndWaitButton.Click += new System.EventHandler(this.m_EndWaitButton_Click);
			// 
			// nhScrollBar1
			// 
			this.nhScrollBar1.Location = new System.Drawing.Point(144, 176);
			this.nhScrollBar1.Name = "nhScrollBar1";
			this.nhScrollBar1.Size = new System.Drawing.Size(176, 17);
			this.nhScrollBar1.TabIndex = 38;
			this.nhScrollBar1.Text = "nhScrollBar1";
			this.nhScrollBar1.ValueChanged += new Nevron.UI.WinForm.Controls.ScrollBarEventHandler(this.nhScrollBar1_ValueChanged);
			// 
			// label1
			// 
			this.label1.Location = new System.Drawing.Point(72, 176);
			this.label1.Name = "label1";
			this.label1.Size = new System.Drawing.Size(64, 23);
			this.label1.TabIndex = 39;
			this.label1.Text = "Speed:";
			this.label1.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
			// 
			// label3
			// 
			this.label3.Location = new System.Drawing.Point(72, 240);
			this.label3.Name = "label3";
			this.label3.Size = new System.Drawing.Size(64, 23);
			this.label3.TabIndex = 41;
			this.label3.Text = "Wait Size:";
			this.label3.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
			// 
			// m_WaitSizeNumeric
			// 
			this.m_WaitSizeNumeric.Location = new System.Drawing.Point(144, 240);
			this.m_WaitSizeNumeric.Name = "m_WaitSizeNumeric";
			this.m_WaitSizeNumeric.Size = new System.Drawing.Size(72, 20);
			this.m_WaitSizeNumeric.TabIndex = 40;
			this.m_WaitSizeNumeric.ValueChanged += new System.EventHandler(this.m_WaitSizeNumeric_ValueChanged);
			// 
			// label5
			// 
			this.label5.Location = new System.Drawing.Point(72, 272);
			this.label5.Name = "label5";
			this.label5.Size = new System.Drawing.Size(64, 23);
			this.label5.TabIndex = 43;
			this.label5.Text = "Step:";
			this.label5.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
			// 
			// m_StepNumeric
			// 
			this.m_StepNumeric.Location = new System.Drawing.Point(144, 272);
			this.m_StepNumeric.Name = "m_StepNumeric";
			this.m_StepNumeric.Size = new System.Drawing.Size(72, 20);
			this.m_StepNumeric.TabIndex = 42;
			this.m_StepNumeric.ValueChanged += new System.EventHandler(this.m_StepNumeric_ValueChanged);
			// 
			// m_IncrementButton
			// 
			this.m_IncrementButton.Location = new System.Drawing.Point(336, 80);
			this.m_IncrementButton.Name = "m_IncrementButton";
			this.m_IncrementButton.TabIndex = 44;
			this.m_IncrementButton.Text = "Increment";
			this.m_IncrementButton.Click += new System.EventHandler(this.m_IncrementButton_Click);
			// 
			// textEdit
			// 
			this.textEdit.Location = new System.Drawing.Point(144, 208);
			this.textEdit.Name = "textEdit";
			this.textEdit.Size = new System.Drawing.Size(176, 18);
			this.textEdit.TabIndex = 45;
			this.textEdit.TextChanged += new System.EventHandler(this.textEdit_TextChanged);
			// 
			// label6
			// 
			this.label6.Location = new System.Drawing.Point(72, 208);
			this.label6.Name = "label6";
			this.label6.Size = new System.Drawing.Size(64, 23);
			this.label6.TabIndex = 46;
			this.label6.Text = "Text:";
			this.label6.TextAlign = System.Drawing.ContentAlignment.MiddleRight;
			// 
			// NWaitingBarUC
			// 
			this.Controls.Add(this.label6);
			this.Controls.Add(this.textEdit);
			this.Controls.Add(this.m_IncrementButton);
			this.Controls.Add(this.label5);
			this.Controls.Add(this.m_StepNumeric);
			this.Controls.Add(this.label3);
			this.Controls.Add(this.m_WaitSizeNumeric);
			this.Controls.Add(this.label1);
			this.Controls.Add(this.nhScrollBar1);
			this.Controls.Add(this.m_EndWaitButton);
			this.Controls.Add(this.m_BeginWaitButton);
			this.Controls.Add(this.label4);
			this.Controls.Add(this.m_OrientationCombo);
			this.Controls.Add(this.m_PaletteButton);
			this.Controls.Add(this.label2);
			this.Controls.Add(this.m_StyleCombo);
			this.Controls.Add(this.m_BorderButton);
			this.Controls.Add(this.nWaitingBar1);
			this.Name = "NWaitingBarUC";
			this.Size = new System.Drawing.Size(424, 336);
			((System.ComponentModel.ISupportInitialize)(this.m_WaitSizeNumeric)).EndInit();
			((System.ComponentModel.ISupportInitialize)(this.m_StepNumeric)).EndInit();
			this.ResumeLayout(false);

		}
		#endregion

		#region Fields

		private System.ComponentModel.Container components = null;

		private Nevron.UI.WinForm.Controls.NWaitingBar nWaitingBar1;
		private System.Windows.Forms.Label label4;
		private Nevron.UI.WinForm.Controls.NComboBox m_OrientationCombo;
		private Nevron.UI.WinForm.Controls.NButton m_PaletteButton;
		private System.Windows.Forms.Label label2;
		private Nevron.UI.WinForm.Controls.NComboBox m_StyleCombo;
		private Nevron.UI.WinForm.Controls.NButton m_BorderButton;
		private Nevron.UI.WinForm.Controls.NButton m_BeginWaitButton;
		private Nevron.UI.WinForm.Controls.NButton m_EndWaitButton;
		private Nevron.UI.WinForm.Controls.NHScrollBar nhScrollBar1;
		private System.Windows.Forms.Label label1;
		private System.Windows.Forms.Label label3;
		private Nevron.UI.WinForm.Controls.NNumericUpDown m_WaitSizeNumeric;
		private System.Windows.Forms.Label label5;
		private Nevron.UI.WinForm.Controls.NNumericUpDown m_StepNumeric;
		private Nevron.UI.WinForm.Controls.NTextBox textEdit;
		private System.Windows.Forms.Label label6;
		private Nevron.UI.WinForm.Controls.NButton m_IncrementButton;

		#endregion
	}
}
